src/BuildScriptGenerator/Contracts/IDetector.cs

```diff
@@ -0,0 +1,22 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    /// <summary>
+    /// An abstraction over the detection of all platforms in a given repository.
+    /// </summary>
+    public interface IDetector
+    {
+        /// <summary>
+        /// Returns all platforms detected for the given repository context.
+        /// </summary>
+        /// <param name="ctx">The <see cref="RepositoryContext"/>.</param>
+        /// <returns>A dictionary mapping the detected platforms to the version.</returns>
+        IDictionary<IProgrammingPlatform, string> GetAllDetectedPlatforms(RepositoryContext ctx);
```

Comment:
Did you rebase your branch on top of master? I ask this because you can then test this api usage experience as an end user would do..for example, have you tried using this api from within a console app (i mean a console app outside Oryx) and how is your experience like...for example, how easy is the api to use, is it exposing lot of things which are not really required to,